frappe/public/js/frappe/form/form.js

```diff
@@ -1942,10 +1942,9 @@ frappe.ui.form.Form = class FrappeForm {
 	setup_docinfo_change_listener() {
 		let doctype = this.doctype;
 		let docname = this.docname;
-		let listener_name = `update_docinfo_for_${doctype}_${docname}`;
-		// to avoid duplicates
-		frappe.realtime.off(listener_name);
-		frappe.realtime.on(listener_name, ({ doc, key, action = "update" }) => {
+
+		frappe.socketio.docinfo_subscribe(doctype, docname);
+		frappe.realtime.on("docinfo_update", ({ doc, key, action = "update" }) => {
```

Comment:
since this event is same for all forms now, this handler should only be "attached" once. needs redesign.

Reply:
Fixed in 3a8fa6c

Reply:
fix won't work as expected. the vars `this`, `doctype` and `docname` will always refer to the current object if I'm not mistaken.

Reply:
Yep, `this` was a mess 😄. Instead, I've now gone back to resetting and attaching a new callback. To ensure we aren't showing stale data, I'm popping the old document from locals too (desk makes a getdoc request then) [via https://github.com/frappe/frappe/pull/18875/commits/2b7e4554c4e0dc6efeffecb5e699e1a666388190]